public/app/features/explore/QueryRow.tsx

```diff
@@ -76,8 +76,10 @@ export class QueryRow extends PureComponent<QueryRowProps, QueryRowState> {
     this.props.addQueryRow(exploreId, index);
   };
 
-  onClickClearButton = () => {
-    this.onChange(null, true);
+  onClickToggleHiddenQuery = (query: DataQuery) => {
+    const { exploreId, index } = this.props;
+    !!this.props.query.hide ? (query.hide = false) : (query.hide = true);
```

Comment:
We are mutating props give to us by parent here which can create lots of problems. Better just create a new query and pass that up:
```
const newQuery = {
  ...this.props.query
  hide: !this.props.query.hide
};
this.props.changeQuery(exploreId, newQuery, index, true);
```

This also means you do not have to pass `query` arg here as it always works on `this.props.query`. Also in general it is confusing when we use in the same context both `query` arg and `this.props.query` which both point to the same object.

Reply:
Yes. Makes so much sense. Thanks Andrej. It is fixed now. 🙂